examples/hwapi/hwconfig_pyboard.py: as requested in issue #2832

kfricke opened the pull request:
as requested in issue #2832

dpgeorge commented on examples/hwapi/hwconfig_pyboard.py:
```diff
@@ -0,0 +1,4 @@
+from machine import Pin, Signal
+
+# pyboard has blue LED on pin 'B4', active high (not inverted)
+LED = Signal(Pin('B4', Pin.OUT))
```
Better to use the named pin "LED_BLUE" because that's guaranteed to give you the blue LED on all current and future pyboards.

There are also 3 other LEDs and 1 button, so could add those following hwconfig_dragonboard410c.py

kfricke replied:
rgr, willdo (tonight)

kfricke replied:
Does X17 (the USR button pin) also have a better alternative name?

kfricke commented:
New version with suggested alternative names and respecting the interface to the actual hwapi examples.

dpgeorge commented:
Thanks for the update, just a few additional things:
- I'd number the LEDs with red being first and blue being last, because that's how pyb.LED works.
- The button needs a pull-up enabled on the pin to work, in the Pin constructor.

But notice that the button_reaction.py script doesn't work.  The BUTTON object (a Signal instance) doesn't work with the machine.time_pulse_us function because stmhal does not implement generic pin objects (in the interest of efficiency for software SPI and I2C).  Anyway, that's not a shop stopper to merge this after the above changes are made.

pfalcon commented:
> The BUTTON object (a Signal instance) doesn't work with the machine.time_pulse_us function

That's why none of the existing hwconfig*.py defines button as a Signal. This hwconfig should be consistent with the rest, and further features will be addressed incrementally.

dpgeorge commented:
> This hwconfig should be consistent with the rest, and further features will be addressed incrementally.

Ok, but if it's just a Pin then then the button_reaction demo doesn't work as intended (it times how long you don't press the button, not how long you do).

This really points to a big underlying issue: do we want to make Signal usable in all cases where a Pin is, eg in time_pulse_us, software I2C and SPI, etc?  If no then why bother with Signal.  If yes then we need to work out a way for it to be efficient and then make all ports support it.

pfalcon commented:
> Ok, but 

"further features will be addressed incrementally."

> This really points to a big underlying issue: do we want to make Signal usable in all cases where a Pin is, eg in time_pulse_us, software I2C and SPI, etc?

We already discussed that no, we don't want to make it such (or that would be big argument that Pin and Signal should be one class). We don't want it for I2C, SPI. time_pulse_us is a grey area. I initially though that yes, it should account for Signal, and created button_reaction.py sample. I now have reservations. It's not yet time to discuss them, sorry. Let's please avoid "all at once" approach. Even once it's mentioned the orderly process goes astray: "interesting" proposals for dropping the core API calls appear in the topic ticket, offtopic discussion happens in this ticket, etc.

kfricke commented:
Okay... I will reorder the LEDs by the colors in the LED class enumeration.

But you do not want the push button to be wrapper with a Signal instance, because this kicks of a discussion/design decision you do not want to tackle now or at all, right?

kfricke commented:
Updated as suggested.

dpgeorge commented:
@kfricke I think it's best to remove BUTTON altogether until it's understood how to implement it properly.  Implemented as a Pin, then button_led doesn't work, implemented as a Signal, then button_reaction doesn't work.

kfricke committed:
examples/hwapi/hwconfig_pyboard.py: as requested in issue #2832
- Reordered pin names according to numbering of LEDs on the pyboard LED classes.

kfricke commented:
> ...remove BUTTON altogether...

Check

dpgeorge commented:
Merged in c8febe631af902de0878c48820346fac87acecd4 (after reordering the comments).  Thanks @kfricke.
